kernel/src/path.rs

```diff
@@ -518,4 +600,58 @@ mod tests {
             .unwrap();
         ParsedLogPath::try_from(log_path).expect_err("non-numeric hi");
     }
+
+    #[test]
+    fn test_empty_filename() {
+        let table_log_dir = table_log_dir_url();
+
+        // Test URL ending with a slash
+        let log_path = table_log_dir.join("").unwrap();
+        let result = ParsedLogPath::try_from(log_path);
+        assert!(result.is_err(), "Expected an error for empty filename");
+        assert!(
+            matches!(result, Err(Error::InvalidLogPath(_))),
+            "Expected InvalidLogPath error for empty filename"
+        );
+    }
+
+    #[test]
+    fn test_ok_none_and_unknown_cases() {
+        let table_log_dir = table_log_dir_url();
+
+        // Test cases for non-numeric version (should return Ok(None))
+        let test_cases = vec!["_last_checkpoint", "abc.json", "version_001.json"];
+
+        for file_name in test_cases {
+            let log_path = table_log_dir.join(file_name).unwrap();
+            let result = ParsedLogPath::try_from(log_path);
+            assert!(
+                matches!(result, Ok(None)),
+                "Expected Ok(None) for non-numeric version: {}",
+                file_name
+            );
+        }
+
+        // Test case for missing file extension (should return Ok(None))
+        let log_path = table_log_dir.join("00000000000000000010").unwrap();
+        let result = ParsedLogPath::try_from(log_path);
+        assert!(
+            matches!(result, Ok(None)),
+            "Expected Ok(None) for missing file extension"
+        );
+
+        // Test case for empty extension (should return Ok(Some) with Unknown file type)
+        let log_path = table_log_dir.join("00000000000000000011.").unwrap();
```

Comment:
Nice catch. I didn't even think of this case when first writing these tests.

Reply:
Retained this.